### PR TITLE
Support both 'buf mod push' for v1.0.0^

### PR DESCRIPTION
**Pull request by amckinney**
Now that we've moved commands around for `v1`, this adapts the action to support both the new and old way of pushing modules to the BSR.

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -99,3 +125,26 @@ async function runPush(): Promise<null|Error> {
...
     return null;
 }
+
+// bufVersion returns the version of the buf binary.
+function bufVersion(binaryPath: string): string | Error {
+    let version = '';
+    try {
+      version = cp.execSync(`${binaryPath} version`).toString();
+    } catch {
+      core.info(`${binaryPath} does not support the 'version' sub-command`)
```
Comment (bufdev) [inline src/run.ts:135]: is this just logging? will this show up to the user at all?

File: src/run.ts [hunk 2]
```diff
@@ -81,14 +87,34 @@ async function runPush(): Promise<null|Error> {
         };
     }
 
+    const version = bufVersion(binaryPath);
```
Comment (bufdev) [inline src/run.ts:90]: does this for for `-dev` etc?

**Comment by amckinney** [comment]
Discussed offline - if we are going to change the commands we use for `v1`, we don't need to support _all_ versions of `buf` in the latest `buf-push-action` release. Instead, we can require that `buf` is at least `v1` and only support `buf mod push` and `buf version`.

The way forward will depend on whether or not we support `buf push` and `buf --version` for the long term. More details can be found [here](https://github.com/bufbuild/buf/issues/483#issuecomment-919587583).

**Comment by amckinney** [comment]
This isn't necessary anymore now that we're using `buf push` and `buf --version`.

**Pull request closed without merging**